<a href="https://colab.research.google.com/github/arechigajd/strokes/blob/main/strokes.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import pandas as pd

In [ ]:
filename = ('/content/healthcare-dataset-stroke-data.csv')
stroke_df = pd.read_csv(filename)
stroke_df.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

In [ ]:
stroke_df.shape

(5110, 12)

1. The source of this dataset is Kaggle
2. The dataset is about strokes
3. The target is a stroke and whether it will happen or not
4. This is a classification problem
5. 11 features
6. 5110 rows of data
7. Challenges I foresee in this data would be to find the best features and preprocess the data correctly with no data leakage.

In [ ]:
path = ('/content/adult.csv')
adin_df = pd.read_csv(path)
adin_df.head()

age  workclass  fnlwgt     education  educational-num      marital-status  \
0   25    Private  226802          11th                7       Never-married   
1   38    Private   89814       HS-grad                9  Married-civ-spouse   
2   28  Local-gov  336951    Assoc-acdm               12  Married-civ-spouse   
3   44    Private  160323  Some-college               10  Married-civ-spouse   
4   18          ?  103497  Some-college               10       Never-married   

          occupation relationship   race  gender  capital-gain  capital-loss  \
0  Machine-op-inspct    Own-child  Black    Male             0             0   
1    Farming-fishing      Husband  White    Male             0             0   
2    Protective-serv      Husband  White    Male             0             0   
3  Machine-op-inspct      Husband  Black    Male          7688             0   
4                  ?    Own-child  White  Female             0             0   

   hours-per-week native-country income  
0              40  United-States  <=50K  
1              50  United-States  <=50K  
2              40  United-States   >50K  
3              40  United-States   >50K  
4              30  United-States  <=50K

In [ ]:
adin_df.shape

(48842, 15)

1. Source of the data set is Kaggle
2. The data is about the income of adults based on certain factors
3. The target is the income and how certain factors will affect future income of adults
4. This is a classification problem since income is divided into 2 categories, <=50K and >50K
5. 14 features
6. 48842 rows of data
7. The challenge I foresee is filtering out all the necessary data from the unnecessary data to have a fast, well-balanced model.